speech/api/README.md

```diff
@@ -38,25 +38,42 @@ See the
 [Cloud Platform Auth Guide](https://cloud.google.com/docs/authentication#developer_workflow)
 for more information.
 
+### Pre-requisites
+
+Below is the list of pre-requisites you need to perform
+
+* Install [gcloud][gcloud]
+* Install [pip][pip]
+
 ### Install the dependencies
 
-The sample uses the [PyAudio][pyaudio] library to stream audio from your computer's microphone.  PyAudio depends on [PortAudio][portaudio], which may need to be compiled when you install PyAudio. If you run into compilation issues that mention PortAudio, you may have to [install some dependencies][pyaudio-install].
+The sample uses the [PyAudio][pyaudio] library to stream audio from your computer's microphone.  PyAudio depends on [PortAudio][portaudio], which may need to be compiled when you install PyAudio. 
+
+* If you run into compilation issues that mention PortAudio, you may have to [install some dependencies][pyaudio-install].
+
+* If you run into compilation issues that mention PortAudio for Ubuntu 
+
+  ```
+  $ sudo apt-get install portaudio19-dev
+  ```
 
 * If you're running the `speech_rest.py` sample:
 
     ```sh
-    $ sudo pip install -r requirements-speech_rest.txt
+    $ sudo pip install -r requirements-speech_rest.txt --ignore-installed six
```

Comment:
I have been thinking about that as well. virtual env or conda. But, the problem is what if they want to do something outside of virtual?